game-core/src/main/java/games/strategy/engine/framework/startup/ui/PlayerTypes.java

```diff
@@ -0,0 +1,167 @@
+import games.strategy.triplea.ai.pro.ProAi;
+import games.strategy.triplea.ai.weak.DoesNothingAi;
+import games.strategy.triplea.ai.weak.WeakAi;
+import games.strategy.triplea.settings.ClientSetting;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+import org.triplea.injection.Injections;
+
+@Value
+public class PlayerTypes {
+
+  public static final Type HUMAN_PLAYER =
+      new Type("Human") {
+        @Override
+        public Player newPlayerWithName(final String name) {
+          return new TripleAPlayer(name) {
+            @Override
+            public Type getPlayerType() {
+              return HUMAN_PLAYER;
+            }
+          };
+        }
+      };
+  public static final Type WEAK_AI =
+      new Type("Easy (AI)") {
+        @Override
+        public Player newPlayerWithName(final String name) {
+          return new WeakAi(name);
+        }
+      };
+  public static final Type FAST_AI =
+      new Type("Fast (AI)") {
+        @Override
+        public Player newPlayerWithName(final String name) {
+          return new FastAi(name);
+        }
+      };
+  public static final Type PRO_AI =
+      new Type("Hard (AI)") {
+        @Override
+        public Player newPlayerWithName(final String name) {
+          return new ProAi(name);
+        }
+      };
+  public static final Type DOES_NOTHING_AI =
+      new Type("Does Nothing (AI)") {
+        @Override
+        public Player newPlayerWithName(final String name) {
+          return new DoesNothingAi(name);
+        }
+      };
+  /** A hidden player type to represent network connected players. */
+  public static final Type CLIENT_PLAYER =
+      new Type("Client", false) {
+        @Override
+        public Player newPlayerWithName(final String name) {
+          return new TripleAPlayer(name) {
+            @Override
+            public Type getPlayerType() {
+              return CLIENT_PLAYER;
+            }
+          };
+        }
+      };
+  /** A 'dummy' player type used for battle calc. */
+  public static final Type BATTLE_CALC_DUMMY =
+      new Type("None (AI)", false) {
+        @Override
+        public Player newPlayerWithName(final String name) {
+          throw new UnsupportedOperationException(
+              "Fail fast - bad configuration, should instantiate dummy player "
+                  + "type only for battle calc");
+        }
+      };
+
+  Collection<Type> playerTypes = new ArrayList<>();
+
+  public PlayerTypes() {
+    playerTypes.add(HUMAN_PLAYER);
+    playerTypes.add(WEAK_AI);
+    playerTypes.add(FAST_AI);
+    playerTypes.add(PRO_AI);
+    playerTypes.add(DOES_NOTHING_AI);
+    playerTypes.add(CLIENT_PLAYER);
+    playerTypes.add(BATTLE_CALC_DUMMY);
+
+    final Injections injection = Injections.getInstance();
+    if (injection != null) {
```

Comment:
injections should always be non-null. While it *can* be null, we should be instantiating as early as possible in any main method.

Reply:
`Injections` is null during tests.

Reply:
This is moot now right?

Reply:
The code has been removed.